requirements.txt

```diff
@@ -1,44 +1,49 @@
 # -------------------------------------------------------------------
-# if want to use the pymodbus serial stack, uncomment these
-# -------------------------------------------------------------------
-# pyserial>=3.4
-# -------------------------------------------------------------------
-# if you want to run the tests and code coverage, uncomment these
-# -------------------------------------------------------------------
-#coverage==4.4
-#mock==2.0.0
-#nose==1.3.7
-#pep8==1.7.0
-# -------------------------------------------------------------------
-# if you want to use the Twisted asynchronous version, uncomment these
-# -------------------------------------------------------------------
-#Twisted==17.1.0
-#zope.interface==4.4.0
-#pyasn1==0.2.3
-#wsgiref==0.1.2
-#cryptography==1.8.1
+# Python packages needed for pymodbus runtime.
+# -------------------------------------------------------------------
+# Different pymodbus options need a number of packages are therefore
+# optional and do not need to be installed.
+#
+# pymodbus is only tested on python version 3.8, 3.9 and 3.10
+# depending on the usage profile it might work on other versions.
+python_version>="3.8"
 
-# -------------------------------------------------------------------
-# if you want to use the Tornado asynchronous version, uncomment these
-# -------------------------------------------------------------------
-# tornado==4.5.2
 
 # -------------------------------------------------------------------
-# if you want to use the Asyncio asynchronous version, uncomment these
+# These packages are optional for the runtime, but demanded for
+# document generation, development and test.
+#
+# But not installing them you save a few 100kb of disk space,
+# and nothing else, since they are not loaded unless you use them
+# in your application.
 # -------------------------------------------------------------------
-# pyserial-asyncio==0.4.0;python_version>="3.4"
 
-# -------------------------------------------------------------------
-# if you want to build the documentation, uncomment these
-# -------------------------------------------------------------------
-#Jinja2==2.9.6
-#Pygments==2.2.0
-#Sphinx==1.5.5
-#docutils==0.13.1
-#pydoctor==16.3.0
+# used by REPL.
+click>=6.7
+prompt-toolkit==2.0.4
+
+# used to enable serial communication.
+pyserial>=3.5
+pyserial-asyncio==0.4.0
+
+# used by datastore.
+sqlalchemy>=1.1.15
```

Comment:
We do not have to install these modules unless users explicitly wants to use these features. For e.g if somebody wants to install pymodbus with `REPL` they can simply to `pip install pymodbus[repl]`.

Reply:
Refer setup.py

Reply:
I agree it is not a demand, but it makes everything a lot easier, and what is the drawback...they occupy a bit of space in "site-packages". I wrote in the text that they can be commented out.

Reply:
I am ok with commented out `requirements.txt` for easy reference except for pyserial and pyserial-async packages

Reply:
My preference was to go the other way, install all unless the end-user comments it out. The reason is that many users to not view the content of the files but simply install them.